### ISSUE TITLE
EI: Baneblade Item's Weapon Has a Hidden "Always Use On Defence" Effect That Should be Included in its Description

### ISSUE THREAD

**Issue by IskarJarak01** [wesnoth/wesnoth#8914]
### Describe the desired feature

In Eastern Invasion, the player can gain an item, the Baneblade, which gives a unit an extra melee weapon with the Beserk special.

Normally, a defending unit with a choice of weapons may decide not to use a weapon with Beserk if it has a high chance of being killed.

The Baneblade item has a hidden effect which forces the unit to always use its Beserk weapon on defence by setting the weight of other melee weapons to zero. 
This should be included in the weapon's description as a dummy effect or similar.

Other notes:
There's a minor issue where if another item that gives a melee attack is equipped afterwards (only the Plague Staff item) the unit will still consider using that attack on defence.
Attacks gained from advancements (Horseman -> Knight) are still affected by the Baneblade item and aren't used on defence.

**Comment by Wedge009** [comment]
@Dalas121 Should be straightforward to add the relevant explanation, right?

**Comment by Dalas121** [comment]
Yeah, adding an explanation should be simple. _"This attack will always be used on defense."_

Fixing the attack-order bugs will be trickier; thanks for noticing and bringing this up. Does anyone know of a way to solve this kind of thing?

**Comment by babaissarkar** [comment]
@Dalas121 what's the situation on this? I think this could be dealt with a new special note, but correct me if I'm wrong. Feel free to ask if you need any engine side support.

**Comment by Dalas121** [comment]
Ah, thanks for reminding me about this. I think I've fixed the issue (both issues) in the linked PR.
